### PR TITLE
wallet: reduce coin selection iterations

### PR DESCRIPTION
**Pull request by murchandamus**
Remove coin selection iteration requiring 6 confs
    
It's not clear what purpose we serve with this additional iteration of coin selection that restricts foreign UTXOs to at least six confirmations.
    
If the wallet hasn't been used in a long time, requiring six confirmations or one is equivalent. If the wallet has just received
    funds, the important distinction is still between confirmed and    unconfirmed UTXOs: we haven't had any reorgs of more than a single block    in multiple years, and the ones we had have been benign—the differences    of included transactions between blocks at the same height is generally    minuscule and can be explained by transaction relay latency or the rate    of block template updates. The risk of a UTXO getting reorged out due to    requiring a single confirmation instead of six can be discounted and    doesn't suffice to instigate a full additional coin selection iteration.

### CODE REVIEW

**Comment by 1440000bytes** [comment]
> If the wallet has just received
funds, the important distinction is still between confirmed and unconfirmed UTXOs: we haven't had any reorgs of more than a single block in multiple years, and the ones we had have been benign—the differences of included transactions between blocks at the same height is generally minuscule and can be explained by transaction relay latency or the rate of block template updates.

How will this work in an attack shared in this tweet? https://nitter.net/AchimWar/status/1505980778458484736

> The risk of a UTXO getting reorged out due to requiring a single confirmation instead of six can be discounted and doesn't suffice to instigate a full additional coin selection iteration.

I am not sure if all users will be okay with this discount. Disagree and if it ain't broke, don't fix it.

**Comment by achow101** [comment]
Concept ACK

**Comment by ClaraBara22** [comment]
1440000bytes, why on earth do people have a problem with the discount? I don't even know how much it is, but I think you all work hard enough that whatever you get, you've earned it...fair and square. When I was like a real working person I got paid for every patient I saw. I really do get it. May we live long and prosper.

**Comment by MarcoFalke** [comment]
This can also happen in normal operation, absent any attack, if fee bumping becomes more common. Two competing blocks often contain the same tx with different fee rates (thus different txid). Even if it happens only once a month, I think it can be inconvenient for users to having to re-send a payment, risking double-pay if the new payment gets confirmed in a chain with the (presumed) stale block.

**Review by ghost — CHANGES_REQUESTED** [review]
Concept NACK

> The likelihood of a situation arising in which this is relevant is so minuscule that performing the additional computation step is not justified

This situation looks possible with governments involved in bitcoin, mining, councils etc. We have not disabled firewall in our office because there were no attacks in last few months.

We are going through lot of things that I don't want to share here as it will be offtopic but still affects bitcoin and I expect bitcoin developers to respect it.

> This can also happen in normal operation, absent any attack, if fee bumping becomes more common. Two competing blocks often contain the same tx with different fee rates (thus different txid). Even if it happens only once a month, I think it can be inconvenient for users to having to re-send a payment, risking double-pay if the new payment gets confirmed in a chain with the (presumed) stale block.

This is another point and I appreciate adversarial thinking by maintainers.

tl;dr: I respect PR author's contribution in coin selection and bitcoin however I am not okay with this change

**Comment by luke-jr** [comment]
>we haven't had any reorgs of more than a single block in multiple years, and the ones we had have been benign

But a change like this creates an incentive to do so.

Agree with @1440000bytes - this probably isn't a good idea.

(Maybe if we could automatically detect and handle problems?)

**Comment by kristapsk** [comment]
Agree with @1440000bytes and @luke-jr here, NACK as it is. Probably 6 could be configurable number and then user could set it to 1 if he wants to?

**Comment by Xekyo** [comment]
Okay, thanks for the feedback.
